File: server/models/userrelation.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class userRelation extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      this.belongsTo(models.profiles, { foreignKey: "followerProfileId" });
      this.belongsTo(models.profiles, { foreignKey: "beingFollowedProfileId" });
    }
  }
  userRelation.init({
    followerProfileId: { type: DataTypes.INTEGER, allowNull: false },
    beingFollowedProfileId: { type: DataTypes.INTEGER, allowNull: false },
    uuid: { type: DataTypes.UUID, allowNull: false, defaultValue: DataTypes.UUIDV4 }
  }, {
    sequelize,
    modelName: 'userRelation',
  });
  return userRelation;
};

File: lib/models/user/userrelation.dart
class UserRelation {
  final int id;
  final int followerProfileId;
  final int beingFollowedProfileId;
  final DateTime createdAt;
  DateTime updatedAt;

  UserRelation({
    required this.id,
    required this.beingFollowedProfileId,
    required this.followerProfileId,
    required this.createdAt,
    required this.updatedAt,
  });
}
